### PR TITLE
release/public-v2: b4b reproducibility for restart runs

### PR DESCRIPTION
**Pull request by climbfuji**
## Description

This PR and associated PRs below fix the b4b reproducibility issues for the release/public-v2 branches. With these changes, restart runs are b4b identical to continues runs for regional and global applications with both GFS v15p2 and RRFS v1 alpha physics.

Changes: 
- copied over from develop (and originally contributed by @SMoorthi-emc): bugfixes in io/FV3GFS_io.F90 to achieve restart reproducibility for global runs using the GFS v15p2 suite; see original PRs for develop https://github.com/NOAA-EMC/fv3atm/pull/178 and https://github.com/NOAA-EMC/fv3atm/pull/212
- add necessary variables for NoahMP restart reproducibility to gfsphysics/GFS_layer/GFS_restart.F90 (as described in https://github.com/NCAR/ccpp-physics/issues/367)
- update submodule pointer for ccpp-physics to include PR https://github.com/NCAR/ccpp-physics/pull/519 (release/public-v5: scientific documentation)

## Testing

See https://github.com/ufs-community/ufs-weather-model/pull/417

## Dependencies

https://github.com/NCAR/ccpp-physics/pull/519
https://github.com/NOAA-EMC/fv3atm/pull/246
https://github.com/ufs-community/ufs-weather-model/pull/417

### CODE REVIEW

File: io/FV3GFS_io.F90 [hunk 1]
```diff
@@ -1357,28 +1357,47 @@ subroutine sfc_prop_restart_read (Sfcprop, Atm_block, Model, fv_domain)
...
-          endif
+            Sfcprop(nb)%tsfc(ix)  = Sfcprop(nb)%tsfco(ix)
+            if (Sfcprop(nb)%slmsk(ix) == 1) then
+              Sfcprop(nb)%zorl(ix) = Sfcprop(nb)%zorll(ix)
+              Sfcprop(nb)%tsfc(ix) = Sfcprop(nb)%tsfcl(ix)
+            else
+              tem = one - Sfcprop(nb)%fice(ix)
+              Sfcprop(nb)%zorl(ix) = Sfcprop(nb)%zorli(ix) * Sfcprop(nb)%fice(ix) &
+                                   + Sfcprop(nb)%zorlo(ix) * tem
+              if (Sfcprop(nb)%fice(ix) > min(Model%min_seaice,Model%min_lakeice)) then
```
Comment (SMoorthi-emc) [inline io/FV3GFS_io.F90:1394]: Why is this statement correct?

**Review by SMoorthi-emc — APPROVED** [review]
Nevermind, it is ok.

File: io/FV3GFS_io.F90 [hunk 2]
```diff
@@ -1077,16 +1077,16 @@ subroutine sfc_prop_restart_read (Sfcprop, Atm_block, Model, fv_domain)
 
         if (Sfcprop(nb)%lakefrac(ix) > zero) then
           Sfcprop(nb)%oceanfrac(ix) = zero ! lake & ocean don't coexist in a cell
-          if (Sfcprop(nb)%fice(ix) < Model%min_lakeice) then
-             Sfcprop(nb)%fice(ix) = zero
-             if (Sfcprop(nb)%slmsk(ix) == 2) Sfcprop(nb)%slmsk(ix) = 0
-          endif
+!          if (Sfcprop(nb)%fice(ix) < Model%min_lakeice) then
```
Comment (junwang-noaa) [inline io/FV3GFS_io.F90:1080]: May I ask if these statements commented out are required for b4 restart reproducibility? I am curious why this change is required as it might indicate the fice could be <min_lakeice in restart run.

Comment (SMoorthi-emc) [reply io/FV3GFS_io.F90:1080]: I guess this question is for Dom.
(this code in my branch is slightly different; it never made it  to the dev version).

Comment (climbfuji) [reply io/FV3GFS_io.F90:1080]: I copied over the exact changes that were made in PRs #178 and #212 for develop. I don't think we should try to do something different here than what was done for develop (and what took us - well, actually Moorthi - months to figure out when we finalized the GFS v16 prod equivalent in develop). Please see these two PRs.

Comment (climbfuji) [reply io/FV3GFS_io.F90:1080]: > I guess this question is for Dom.
> (this code in my branch is slightly different; it never made it to the dev version).

Forgot to say, the code changes in this PR are from you originally - as you said, your slightly different version never made it to develop at that time, I remember those back and forth conversations quite well.

**Review by JeffBeck-NOAA — APPROVED** [review]

**Comment by SMoorthi-emc** [comment]
Dom,
  I think some of this code is from Shan as some of my changes did not make in.
Moorthi

**Comment by climbfuji** [comment]
> Dom, I think some of this code is from Shan as some of my changes did not make in. Moorthi
> […](#)

Thanks for the clarification. Nonetheless, I think these changes are good and they definitely solve the problem with the b4b restart reproducibility (the global GFS v15p2 suite runs in release/public-v2 do not give b4b identical results in restart runs without this PR).

**Comment by climbfuji** [comment]
This PR is ready to merge.

**Review by junwang-noaa — APPROVED** [review]
